controller/lib/hal/hal.h

```diff
@@ -586,6 +587,10 @@ inline void HalApi::IntDisable() {}
 inline void HalApi::IntEnable() {}
 inline bool HalApi::IntSuspend() { return false; }
 
+inline uint32_t HalApi::crc32(uint8_t *data, uint32_t length) {
+  return soft_crc32(reinterpret_cast<char *>(data), length);
+}
```

Comment:
I may be missing it, but aren't crc32_init(), crc32_reset, crc32_accumulate, and crc32_get all still unimplemented in the native hal?

Reply:
Yep, missed the init

Reply:
private doesn't mean they don't need an implementation though?

Reply:
I don't understand why would they. They are not a part of external HalApi interface that any Hal client would access. What am I missing here? 

Maybe I should hide them under #ifdef BARE_STM32

Reply:
It'd be like in C declaring function prototypes that you never intend to implement (on a particular platform).  Even if those function prototypes are "private", meaning they're defined in a `.c` file, it's still not a good idea to do so.  When a user tries to call them, they'll get a linker error, and it won't be clear whether they "weren't supposed" to call them or if they're supposed to implement them or what.

If they're private, "users" is restricted to "the hal class", but still.

> Maybe I should hide them under #ifdef BARE_STM32

works for me.

Reply:
it's a late night though. I'll be signing off for now. thanks!